doc/specs/drafts/#3327 - Application Theming/#10509 - Mica.md

```diff
@@ -0,0 +1,232 @@
+If we make enabling Mica for the control a per-profile setting, I believe that
+will lead to greater user confusion. It would result in "spooky action at a
+distance", where creating any pane with Mica would force the entire window to
+have a Mica background. This would change the appearance of any other unblurred
+transparent panes in the window, causing them to also be subjected to the Mica
+treatment as well.
+
+**Proposal**: create a window-level theme property `window.background.useMica`
+(or similar), which will enable Mica for the entire window. When enabled, users
+can use a fully transparent, unblurred background for their profile to achieve
+the Mica effect within the control. When enabled, users **won't** be able to see
+through to the desktop with any vintage opacity settings.
+
+By nesting Mica usage under `"window.background"`, it will be clearer that it's
+something that applies to the whole window, as the background for everything.
+I believe this is the most acceptable way to expose Mica to our users without
+"spooky action at a distance".
+
+An example of what mica in the control area might look like:
+
+![Mica in the TermControl](./mica-in-control-000.png)
+
+### Mica in the titlebar
+
+To achieve Mica in the titlebar, we'll similarly need to allow users to set the
+titlebar area to totally transparent, to allow the mica behind the window to be
+visible. A simple theme to achieve that might look like:
+
+```jsonc
+{
+    "theme": "My Mica Titlebar Theme",
+    "themes": [
+        {
+            "name": "My Mica Titlebar Theme",
+            "window.background.useMica": true, // Use mica behind the window
+            "tabRow.background": "#00000000", // Make the TabView Transparent
+        }
+    ]
+}
+```
+
+As noted before, due to the intentional limitations of the Mica material, if the
+user wants Mica in the tab row, they won't be able to use vintage opacity in any
+controls. The vintage opacity of the controls will show through to the Mica
+layer, instead of transparent through the whole window.
+
+## Considered implementations
+
+* We experimented with a new DWM API in SV2 which should enable us to set the
+  background of our window to Mica. This did seem to work for the root window.
+  It however, did not seem to work for the "drag window", the child HWND which
+  we use to intercept nonclient messages in our titlebar area. Apparently, that
+  API does not work at all for `WS_CHILD` windows, by design. This unfortunately
+  prevents us from allowing Mica only in the titlebar area, without also
+  applying it to the rest of the main window.
+* We considered rolling our own Mica brush to allow us to style individual
+  elements. This didn't seem terribly hard at face value, considering [the Mica
+  recipe] is basically open-source. However, it seems to rely on some
+  closed-source `IGraphicsEffect`s from the `Composition.Effects` namespace.
+  Apparently, those effects are only [implemented in Win2D]. It's unclear at

```

Comment:
These effects are implemented in Direct2D, and Win2D is just a thin wrapper over Direct2D. It's possible to write your own classes for these Direct2D effects to be able to use them without needing to import the whole of Win2D. This is actually what WinUI does to use these effects without making it depend on Win2D.

Reply:
I mean yea, we probably could do that. For ex:
https://github.com/microsoft/Win2D/blob/da3fa1e6caab9df443c43d0f7a211d9d159539f8/winrt/lib/effects/generated/ColorSourceEffect.cpp